src/Protocol/ForgeMods.h

```diff
@@ -0,0 +1,46 @@
+
+// ForgeMods.h
+
+// Data structure listing the Forge mods name and versions.
+
+#pragma once
+
+#include <stddef.h>
+
+// tolua_begin
+class cForgeMods
+{
+public:
+	// TODO: can tolua++ bridge AStringMap (etc.) as a Lua table? or non-fixedsize arrays?
+
+	/** Returns the number of Forge mods. */
+	size_t GetNumMods(void) const;
+
+	/** Returns true if the mod with the given name is installed. */
+	bool HasMod(const AString & name) const;
+
+	/** Returns the version of the mod name given. */
+	const AString & GetModVersion(const AString & name) const;
+
+	/** Returns the name of the Forge mod at the given index. */
+	const AString & GetModNameAt(size_t i) const;
+
+	// tolua_end
+
+	cForgeMods();
+	cForgeMods(const AStringMap & a);
+
+	static const cForgeMods & Unmodded(void);
+
+	bool Add(const AString & a_Name, const AString & a_Version);
+
+	void Remove(const AString & a_Name);
+
+	AStringMap GetMods() { return m_Mods; }
+
+	bool HasMods() { return m_Mods.size() != 0; }
+
+private:
+	AStringMap m_Mods;
+	AStringVector m_ModNames;
+} ;  // tolua_export
```

Comment:
Cool, I didn't know AStringMap had a binding, I only made cForgeMods because I couldn't find out how to bridge AStringMap to Lua (mod ordering is not important). AStringMap would be simpler and preferable. Started removing cForgeMods, but still having trouble making the Lua API function return AStringMap:

If I have after // tolua_begin this declaration:

```c++
	const AStringMap & GetForgeMods(void) const;
```

then it is bridged as "userdata", I think we want it to be a Lua "table", so plugins can do something like:

```lua
function OnLoginForge(Client)
    local mods = Client:GetForgeMods()

    for name, version in pairs(mods) do
        LOG("SampleForgeMod mod " .. name .. " version " .. version)
    end
```

except this fails: Warning: [e74216953bd757eb|15:30:02] LUA: Plugins/SampleForgeMod/main.lua:60: bad argument #1 to 'pairs' (table expected, got userdata)

CallFunction shouldn't be used in the manual bindings should it?

I now see there is a Lua API function manually bound, returning a table: tolua_cFile_GetFolderContents - but on the C++ side returns an AStringVector. Trying similar code but with AStringMap:

```c++
	cLuaState S(L);
...
	S.Push(Client->GetForgeMods());
```

GetUUIDsFromPlayerNames, GetPermissions, GetRestrictions, GetTeamNames, all return an AStringVector (not AStringMap) as a Lua table. Is there no example of a C++ function returning AStringMap bridged to a Lua table?